test/Makefile

```diff
@@ -0,0 +1,17 @@
+REGISTRY ?= artifactory.delivery.puppetlabs.net
+TEST_IMAGE := $(REGISTRY)/cd4pe-config-test-vm:latest
+
+.PHONY: build-test-image push-test-image
+
+build-test-image:
+	cp $(HOME)/.cdpe-workflow-tests-config.json ..
+	docker build \
+		--pull \
+		--file Dockerfile \
+		--tag $(TEST_IMAGE) \
+		..
+	rm ../.cdpe-workflow-tests-config.json
+
+push-test-image:
+	docker login https://$(REGISTRY)
+	docker push $(TEST_IMAGE)
```

Comment:
We could move this build and push information into the README as instructions, if we wanted to not add a Makefile 🤷‍♂

Reply:
Given this is a public repo and this is an internal requirement, I'm not sure. Unless this is how we want the world to use this, assuming we parameterize and/or find a way for general users to provide a params file with the right content.

Reply:
This stuff isn't for public users, so not sure what we should do about that.  Was just saying we could leave this as documentation if we wanted, somewhere, if we didn't want this Makefile for some reason 🤷‍♂

Reply:
My take on it is to put it in docs since its strictly internal. We can always start a local tools repo (assuming we don't have one thats private already). What does @ericwilliamson think?

Reply:
Does "put it in docs" mean in this public README or in Confluence?

Reply:
I think it's fine as it is right now having a Makefile.

However it deos open a bigger question - now taht this repo is open source, does it make sense for this tooling to continue living here, or should we move it into our main repo? Would that make this work easier? Harder? Would the devs keep it more in mind since it's not out of sight?

Reply:
internal only docs. or a private repo for internal tools.

Reply:
@ericwilliamson given the general use case (vs. internal dev of the tool itself), having things move over to the main repo as long as we have a way of getting the module repo itself from a settable location.

Reply:
Yeah we could move this docker stuff elsewhere, since we're not really baking this repo into it.

The `configTestVm.sh` script is the only place where Docker comes in at runtime, and it just pulls the image from Artifactory.

Use of this tooling would still require this repo is cloned and the `configTestVm.sh` script is executed from the `test/` directory.

Reply:
^^^ this ^^^

Reply:
(or we could put back in _as an option_ the ability to pull the module into a tmpdir)

Reply:
Yep totally. We can iterate on where it lives and how it's used.

Important thing first is get everybody using it!